### PR TITLE
[FIX] mass_mailing: ensure test templating is true

### PR DESCRIPTION
**Pull request by madprog**
When using a template with inline placeholders in the body, the Test button will replace the placeholders, but they will remain in the final body when the mass mailing is sent for real.

This commit fixes this issue by using the correct rendering engine for the Test button too.

[OPW-2819032](https://www.odoo.com/web#model=project.task&id=2819032)
[OPW-2828461](https://www.odoo.com/web#model=project.task&id=2828461)

### CODE REVIEW

**Review by tde-banana-odoo — COMMENTED** [review]
Thanks for the PR ! Nice tests, covering both the test and the actual send, for subject and body :) . LGTM :)

**Comment by tde-banana-odoo** [comment]
Just made a tracking task, if you want to link it in commit message (task-2828461), just for followup.

**Comment by madprog** [comment]
@odoo/rd-security Indeed a modified line contains `sanitize=False`, but it already did before this PR! Could you please bypass the error? :grin:

**Comment by madprog** [comment]
Successfully checked that commit 7a3faef05656991f95cfce802a6e92a0cd92d192 with the new test only is failing with the error `AssertionError: 'Hello {{ object.name }} Mitchell Admin' not found in '...'` (see [runbot build 14692059](https://runbot.odoo.com/runbot/build/14692059)), whereas commit 5080ac132e0458fe01ffb69d661f761f27899597 fixes this issue. The last force-pushed commit is a merge of the two.

File: addons/test_mass_mailing/tests/test_mailing_test.py [hunk 1]
```diff
@@ -59,3 +60,49 @@ def test_mailing_test_button(self):
...
+            'contact_ids': [Command.create({
+                'name': 'Mitchell Admin',
+                'email': 'real@real.com',
+            })],
+        })
+
+        mailing = self.env['mailing.mailing'].create({
+            'name': 'TestButton',
+            'subject': 'Subject {{ object.name }} <t t-out="object.name"/>',
+            'preview': 'Preview {{ object.name }} <t t-out="object.name"/>',
```
Comment (madprog) [inline addons/test_mass_mailing/tests/test_mailing_test.py:79]: Hmm, I didn't find a way to easily test the preview, so maybe I should remove this line
```suggestion
```

Comment (madprog) [inline addons/test_mass_mailing/tests/test_mailing_test.py:79]: Hmm, I didn't find a way to easily test the preview, so maybe I should remove this line
```suggestion
```

**Comment by JKE-be** [comment]
@robodoo override=ci/security  sanitize=False was already present and ok...

File: addons/test_mass_mailing/tests/test_mailing_test.py (hunk 1 above)
Comment (tde-banana-odoo) [reply addons/test_mass_mailing/tests/test_mailing_test.py:79]: Preview is already tested just above so that's not mandatory to re-add tests about it.

**Comment by JKE-be** [comment]
re @robodoo override=ci/security

**Comment by tde-banana-odoo** [comment]
@robodoo r+

Thanks for the PR :)

**Pull request closed without merging**